app/controllers/annotations_controller.rb

```diff
@@ -17,22 +17,79 @@ class AnnotationsController < ApplicationController
   # POST /:course/annotations.json
   action_auth_level :create, :course_assistant
   def create
-    annotation = @submission.annotations.new(annotation_params)
+    # primary annotation
+    primary_annotation = @submission.annotations.new(annotation_params)
 
-    ActiveRecord::Base.transaction do
-      annotation.save
-      annotation.update_non_autograded_score
+    if @submission.group_key.empty?
```

Comment:
seems like based on this implementation, you are:
- adding a group key
- creating a duplicate annotation related to the primary annotation via the group key
- and then deleting and updating all the group key.

I was under the assumption that the implementation be something like:
- when creating an annotation for a group add a foreign key that is the group_id
- when viewing a group submission, retrieve all annotations based on the group_id as well

I think the current implementation works for now, but I'm afraid with future additions it might get confusing. For example, whenever we have something similar to a shared_comments feature, we need to also ensure that the duplicates do not have the shared_comment field checked.

Reply:
1. The current implementation already checks for duplicate shared_comment field across annotations.
2. I'm assuming you still want the group key to be present for the same iteration of submissions? Retrieving annotations by group_id gives you annotations for all submissions of the group. Without the group key, it can be hard to figure out which submissions are associated in the same iteration. Versions can be different across group members, and timestamps might differ depending on the database's performance.

Reply:
I can see that it already checks for duplicate shared_comment. What I was trying to say that if we need to introduce a new feature similar to shared_comment, we need to think about this particular edge case and update group comments accordingly, which might get tricky.

My main thought is that it would be best if there can be a way for us to only store 1 single copy of annotation, without needing to create duplicates.

Reply:
I would propose retaining the submission group_key and add group_id to annotations. Should we disable annotating on non-latest submission in the future, we can get rid of group_key as well. Let me know if it sounds ok!